crates/ra_ide/src/syntax_highlighting.rs

```diff
@@ -623,6 +644,22 @@ fn highlight_element(
                         HighlightTag::SelfKeyword.into()
                     }
                 }
+                T![ref] => {
+                    let modifier: Option<HighlightModifier> = (|| {
+                        let bind_pat = element.parent().and_then(ast::BindPat::cast)?;
+                        if sema.is_unsafe_pat(&ast::Pat::BindPat(bind_pat)) {
+                            Some(HighlightModifier::Unsafe)
+                        } else {
+                            None
+                        }
+                    })();
```

Comment:
Self-invoking closures are neat, but is there perhaps a way to write this simpler, without deploying intricate language machinery?